lib/contentTypeParser.js

```diff
@@ -178,6 +179,10 @@ function rawBody (request, reply, options, parser, done) {
 }
 
 function defaultJsonParser (req, body, done) {
+  if (body === '' || body == null) {
+    return done(new FST_ERR_CTP_EMPTY_JSON_BODY(), undefined)
+  }
+
```

Comment:
@mcollina unfortunately the following block would give a cryptic parse error like `Unexpected token i in JSON at position 2`.

Reply:
It's not a Node core problem, is how `JSON.parse` is implemented in the JS engine.
The current solution is fine :)

Reply:
I like the idea to add this error message. The error is meaningful and doesn't intercept with other error cases. I propose to add it. Something like this makes the difference :)